library/CMakeLists.txt

```diff
@@ -82,6 +82,18 @@ set(MAIN_SOURCES
     RemoteTools.cpp
 )
 
+if(BUILD_TESTING)
+    # TODO Make a function or macro for this
+    add_executable(MiscUtils.test MiscUtils.test.cpp)
+    target_link_libraries(MiscUtils.test dfhack)
+    add_test(NAME MiscUtils.test COMMAND MiscUtils.test)
+
+    # How to get `test` to ensure everything is up to date before running
+    # tests? This add_dependencies() fails with:
+    #   Cannot add target-level dependencies to non-existent target "test".
+    #add_dependencies(test MiscUtils.test)
```

Comment:
when does the `test` target get defined? can we include things in a different order in the top level CMakeLists.txt file to fix this?